pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -38,7 +46,108 @@ func (srv PrometheusSrv) RouteGetAlertStatuses(c *models.ReqContext) response.Re
 }
 
 func (srv PrometheusSrv) RouteGetRuleStatuses(c *models.ReqContext) response.Response {
-	recipient := c.Params(":Recipient")
-	srv.log.Info("RouteGetRuleStatuses: ", "Recipient", recipient)
-	return response.Error(http.StatusNotImplemented, "", nil)
+	ruleResponse := apimodels.RuleResponse{
+		DiscoveryBase: apimodels.DiscoveryBase{
+			Status: "success",
+		},
+		Data: apimodels.RuleDiscovery{},
+	}
+
+	ruleGroupQuery := ngmodels.ListOrgRuleGroupsQuery{
+		OrgID: c.SignedInUser.OrgId,
+	}
+	if err := srv.store.GetOrgRuleGroups(&ruleGroupQuery); err != nil {
+		ruleResponse.DiscoveryBase.Status = "error"
+		ruleResponse.DiscoveryBase.Error = fmt.Sprintf("failure getting rule groups: %s", err.Error())
+		ruleResponse.DiscoveryBase.ErrorType = apiv1.ErrServer
+		return response.JSON(http.StatusInternalServerError, ruleResponse)
+	}
+
+	for _, groupId := range ruleGroupQuery.Result {
+		alertRuleQuery := ngmodels.ListRuleGroupAlertRulesQuery{OrgID: c.SignedInUser.OrgId, RuleGroup: groupId}
+		if err := srv.store.GetRuleGroupAlertRules(&alertRuleQuery); err != nil {
+			ruleResponse.DiscoveryBase.Status = "error"
+			ruleResponse.DiscoveryBase.Error = fmt.Sprintf("failure getting rules for group %s: %s", groupId, err.Error())
+			ruleResponse.DiscoveryBase.ErrorType = apiv1.ErrServer
+			return response.JSON(http.StatusInternalServerError, ruleResponse)
+		}
+
+		newGroup := &apimodels.RuleGroup{
+			Name:           groupId,
+			File:           "", // TODO: set this once we are setting it somewhere
+			LastEvaluation: time.Time{},
+			EvaluationTime: 0, // TODO: see if we are able to pass this along with evaluation results
+		}
+		for _, rule := range alertRuleQuery.Result {
+			instanceQuery := ngmodels.ListAlertInstancesQuery{
+				DefinitionOrgID: c.SignedInUser.OrgId,
+				DefinitionUID:   rule.UID,
+			}
+			if err := srv.store.ListAlertInstances(&instanceQuery); err != nil {
+				ruleResponse.DiscoveryBase.Status = "error"
+				ruleResponse.DiscoveryBase.Error = fmt.Sprintf("failure getting alerts for rule %s: %s", rule.UID, err.Error())
+				ruleResponse.DiscoveryBase.ErrorType = apiv1.ErrServer
+				return response.JSON(http.StatusInternalServerError, ruleResponse)
+			}
+
+			alertingRule := apimodels.AlertingRule{
+				State:       "inactive",
+				Name:        rule.Title,
+				Query:       "", // TODO: get this from parsing AlertRule.Data
```

Comment:
I'm afraid you cannot parse the `AlertRule.Data`. You need to pass the whole `AlertRule.Data` together with the `AlertRule.Condition`